### PR TITLE
K8SPXC-335 Add tests for users rotation and sync

### PR DESCRIPTION
**Pull request by nonemax**
[K8SPXC-335](https://jira.percona.com/browse/K8SPXC-335)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| recreate | passed |
| scaling | passed |
| big-data | passed |
| self-healing | failed |
| limits | passed |
| scaling-proxysql | passed |
| self-healing-advanced | passed |
| monitoring | failed |
| restore-to-encrypted-cluster | failed |
| monitoring-2-0 | failed |
| demand-backup | failed |
| upgrade | failed |
| operator-self-healing | failed |
| affinity | passed |
| upgrade-consistency | passed |
| scheduled-backup | failed |
| security-context | failed |
| one-pod | failed |
| auto-tuning | failed |
| proxysql-sidecar-res-limits | failed |
| demand-backup-encrypted-with-tls | failed |
| users | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| monitoring | failed |
| recreate | passed |
| restore-to-encrypted-cluster | failed |
| monitoring-2-0 | failed |
| demand-backup | passed |
| affinity | passed |
| one-pod | passed |
| scheduled-backup | failed |
| auto-tuning | failed |
| proxysql-sidecar-res-limits | failed |
| users | failed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| restore-to-encrypted-cluster | failed |
| storage | passed |
| self-healing | failed |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| init-deploy | passed |
| limits | passed |
| monitoring | failed |
| big-data | passed |
| monitoring-2-0 | failed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | failed |
| upgrade-consistency | passed |
| security-context | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| proxysql-sidecar-res-limits | failed |
| demand-backup | passed |
| scheduled-backup | failed |
| users | failed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| scaling | passed |
| scaling-proxysql | passed |
| storage | passed |
| upgrade | passed |
| self-healing | passed |
| upgrade-consistency | passed |
| security-context | passed |
| recreate | passed |
| restore-to-encrypted-cluster | failed |
| init-deploy | passed |
| limits | passed |
| monitoring | failed |
| monitoring-2-0 | failed |
| demand-backup | passed |
| scheduled-backup | failed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup-encrypted-with-tls | passed |

File: e2e-tests/functions [hunk 1]
```diff
@@ -858,3 +881,10 @@ start_minio() {
         /usr/bin/env AWS_ACCESS_KEY_ID=some-access-key AWS_SECRET_ACCESS_KEY=some-secret-key AWS_DEFAULT_REGION=us-east-1 \
         /usr/bin/aws --endpoint-url http://minio-service:9000 s3 mb s3://operator-testing
 }
+
+patch_secret() {
+    local secret=$1
+    local key=$2
+    local value=$3
+    kubectl patch secret $secret -p="{\"data\":{\"$key\": \"$value\"}}"
+}
```
Comment (cap1984) [inline e2e-tests/functions:890]: style

File: e2e-tests/users/run [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+compare_mysql_cmd "select-4" "SHOW TABLES;" "-h $cluster-proxysql -uoperator -p$newpass"
+
+desc 'test new users sync'
+run_mysql \
+        "CREATE USER 'testsync'@'%' IDENTIFIED BY '$newpass';" \
+        "-h $cluster-pxc -uroot -p$newpass"
+sleep 15
+compare_mysql_cmd "select-4" "SHOW TABLES;" "-h $cluster-proxysql -utestsync -p$newpass"
+
+destroy "${namespace}"
```
Comment (cap1984) [inline e2e-tests/users/run:61]: style

Comment (nonemax) [reply e2e-tests/users/run]: https://github.com/percona/percona-xtradb-cluster-operator/pull/467/commits/b63cb427878b053ab07c750b2832c25b9288a820

File: e2e-tests/functions (hunk 1 above)
Comment (nonemax) [reply e2e-tests/functions:891]: https://github.com/percona/percona-xtradb-cluster-operator/pull/467/commits/b63cb427878b053ab07c750b2832c25b9288a820

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| recreate | passed |
| big-data | passed |
| scaling | passed |
| storage | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| limits | passed |
| upgrade | passed |
| restore-to-encrypted-cluster | failed |
| self-healing-advanced | failed |
| demand-backup | failed |
| upgrade-consistency | passed |
| monitoring | failed |
| security-context | passed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| scheduled-backup | failed |
| affinity | passed |
| one-pod | failed |
| auto-tuning | passed |
| demand-backup-encrypted-with-tls | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | failed |
| recreate | passed |
| restore-to-encrypted-cluster | failed |
| init-deploy | passed |
| big-data | passed |
| limits | passed |
| monitoring | passed |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| security-context | passed |
| demand-backup | failed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | failed |
| scheduled-backup | failed |
| operator-self-healing | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| self-healing | passed |
| limits | passed |
| self-healing-advanced | passed |
| monitoring | passed |
| scaling | passed |
| monitoring-2-0 | passed |
| scaling-proxysql | passed |
| affinity | passed |
| upgrade | passed |
| one-pod | failed |
| upgrade-consistency | passed |
| security-context | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| big-data | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| operator-self-healing | passed |
| demand-backup | passed |
| scheduled-backup | failed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| storage | passed |
| affinity | passed |
| self-healing | passed |
| one-pod | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| scheduled-backup | failed |
| big-data | passed |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| security-context | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup-encrypted-with-tls | passed |

**Pull request merged**